src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -57,6 +57,11 @@ def load_arguments(self, _):
         type=int,
         help='Maximum number of items to return. Must be a positive integer. Default to 100.'
     )
+    retention_days_arg_type = CLIArgumentType(
+        options_list=['--retention-days'],
+        type=int,
+        help='Number of days to retain the soft delete enabled App Configuration after deleting. Must be a positive integer.'
```

Comment:
> Must be a positive integer. [](http://example.com/codeflow?start=96&length=28)

Can we specify that it should be between 0-7?

Reply:
We want the CLI to not be very tightly coupled to the server. If the server changes the range to be 0-30 days, we don't want to have to update the CLI for it.

Reply:
The flip side of that argument is that right now we know that the server will throw an error if a store is created with more than 7 retention days. Shouldn't we try to avoid the extra server call and add argument validation on client side?
This is something I dont expect will change very often on the server side. We might need to change the validation logic just once when server changes the range to be 0-30 days.

Reply:
I think we would still be in problem because server could change the validation logic any time, but to release a fix in CLI we need to follow Azure CLI timelines.
If it's just CLI, we could match the time when server releases any changes to CLI release timelines, but if multiple clients need to change and have different timelines, then we won't be able to match the release timelines of all clients to the server.

If it is just a message, and no validation, we can still do it as the effects wouldn't be as bad as not being able to provide large retention period from CLI when the server allows it.

I think having the server validate is good option.
We would be documenting this retention period anyways, so users would possibly know acceptable values.
Even if they don't know, making one extra network call is a reasonable tradeoff between potentially having inconsistent behavior between server and CLI.

Reply:
Okay, sounds good.